utils/helpers.js

```diff
@@ -34,34 +44,59 @@ export const sendPostRequest = async (url, { arg }) => {
   return id;
 };
 
-export const sendPatchRequest = async (url, exercise) => {
+export const sendPatchRequest = async (url, payload) => {
   const response = await fetch(url, {
     method: "PATCH",
     headers: {
       "Content-Type": "application/json",
     },
-    body: JSON.stringify(exercise),
+    body: JSON.stringify(payload),
   });
   if (!response.ok) {
     console.error("Couldn't send request");
   }
 };
 
-export const sendPatchRequestCurrentTemplate = async (url, planId) => {
+export const sendPutRequest = async (url, payload) => {
   const response = await fetch(url, {
-    method: "PATCH",
+    method: "PUT",
     headers: {
       "Content-Type": "application/json",
     },
-    body: JSON.stringify(planId),
+    body: JSON.stringify(payload),
   });
   if (!response.ok) {
-    console.error("Setting current tamplate failed");
+    console.error("Couldn't send delete request");
   }
 };
 
+export const createUpdateDelete = async (planId, data, mode) => {
+  if (mode === "isEdit") {
```

Comment:
No need to change anything here, i just want to propose a possible alternative. It feels like there is a little bit of repetition in the current code (not always a bad thing). One possible way to reduce it would be to create a "dictionary" for the different "modes". 

``` js
const modes = {
  isEdit: "isEdit",
  isCreate: "isLog",
  isDelete: "isDelete",
  intialWeeklyLog: "initialWeeklyLog",
};

const url = `/api/plans/${planId}?${modes[mode]}=true&id=${planId}`;
await sendPatchRequest(url, data);
```

Reply:
Oh man, that's way cooler! Thanks.